Thanks, so the changes did look promising, but at least the ones about possibly-unsupported compiler warnings need some more work, as the builds no longer ignore some misplaced flags that they did before (something we might impact with more careful coding of the numerous `*FLAGS` env-/make-vars) and no longer ignore transgressions by third-party code (not much we can do about), e.g.:

* third-party header mess:
```
In file included from phoenixcontact_modbus.c:26:
/usr/include/modbus/modbus.h:157:16: error: identifier '_modbus' is reserved because it starts with '_' at global scope [-Werror,-Wreserved-identifier]
  157 | typedef struct _modbus modbus_t;
      |                ^
```

* system-provided header mess:
```
upsclient.c:1496:5: error: identifier '__i' is reserved because it starts with '__' [-Werror,-Wreserved-identifier]
                                FD_ZERO(&fds);
                                ^
/usr/include/sys/select.h:88:26: note: expanded from macro 'FD_ZERO'
#define FD_ZERO(fdsetp)         __FD_ZERO (fdsetp)
                                ^
/usr/include/bits/select.h:27:18: note: expanded from macro '__FD_ZERO'
    unsigned int __i;                                                         \
                 ^
```

* linker options in compiler commands (might be faults of either our hacks in m4 or `c_build.sh` to get non-default library locations usable, or of system's pkg-config data):
  * OmniOS:
```
configure:21152: WARNING: Optional C library routine strlwr not found
configure:21157: checking for strsep(s1,s2)
configure:21217: /usr/lib/ccache/clang-17 -o conftest  -I/tmp/jenkins-nut/workspace/nut_nut_PR-3489/tmp/include -std=gnu11 -m32 -Werror -Werror=implicit-function-declaration   -I/tmp/jenkins-nut/workspace/nut_nut_PR-3489/tmp/include -L/tmp/jenkins-nut/workspace/nut_nut_PR-3489/tmp/lib -m32 -R/opt/ooce/lib conftest.c  >&5
error: unknown remark option '-R/opt/ooce/lib' [-Werror,-Wunknown-warning-option]
configure:21217: $? = 1
configure: program exited with status 1
configure: failed program was:
...
```
  * NetBSD:
```
configure:21855: result: no
configure:21863: WARNING: Optional C library routine strsep not found
configure:21877: checking for strstr(s1,s2)
configure:21939: /usr/lib/ccache/clang-18 -o conftest -isystem /usr/pkg/include  -I/var/shm/jenkins-nutci-netbsd-92-amd64/nut_nut_PR-3489@3/tmp/include -std=gnu11 -m64 -I/usr/pkg/include -Werror -Werror=implicit-function-declaration   -I/var/shm/jenkins-nutci-netbsd-92-amd64/nut_nut_PR-3489@3/tmp/include -L/var/shm/jenkins-nutci-netbsd-92-amd64/nut_nut_PR-3489@3/tmp/lib -m64 -R/usr/pkg/lib conftest.c  >&5
error: unknown remark option '-R/usr/pkg/lib' [-Werror,-Wunknown-warning-option]
configure:21939: $? = 1
configure: program exited with status 1
configure: failed program was:
...
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -137,13 +137,13 @@ install-fanout-cleanup: install-recursive @dotMAKE@
 touch-include-all-nut_version-generated.timestamp:
 	@[ -s include/nut_version.h ]
 	@touch -r include/nut_version.h -d '-10 seconds' include/.all.nut_version-generated.timestamp 2>/dev/null && exit ; \
-	 touch -d '1970-01-01T00:00:00' include/.all.nut_version-generated.timestamp && exit ; \
+	 touch -t '197001010000.00' include/.all.nut_version-generated.timestamp && exit ; \
 	 touch include/.all.nut_version-generated.timestamp
 
 touch-clients-all-libupsclient_version-generated.timestamp:
 	@[ -s clients/libupsclient-version.h ]
 	@touch -r clients/libupsclient-version.h -d '-10 seconds' clients/.all.libupsclient_version-generated.timestamp 2>/dev/null && exit ; \
-	 touch -d '1970-01-01T00:00:00' clients/.all.libupsclient_version-generated.timestamp && exit ; \
+	 touch -t '197001010000.00' clients/.all.libupsclient_version-generated.timestamp && exit ; \
 	 touch clients/.all.libupsclient_version-generated.timestamp
 
 touch-docs-man-all-linkman-generated.timestamp:
@@ -153,7 +153,7 @@ touch-docs-man-all-linkman-generated.timestamp:
 	 else \
 	    touch -r docs/man/linkman-driver-names.txt -d '-10 seconds' docs/man/.all.nut_linkman-generated.timestamp 2>/dev/null && exit ; \
 	 fi ; \
-	 touch -d '1970-01-01T00:00:00' docs/man/.all.nut_linkman-generated.timestamp && exit ; \
+	 touch -t '197001010000.00' docs/man/.all.nut_linkman-generated.timestamp && exit ; \
 	 touch docs/man/.all.nut_linkman-generated.timestamp
 
 # Verbosity for fanout rule tracing; 0/1 (or "default" that may auto-set

--- a/configure.ac
+++ b/configure.ac
@@ -525,6 +525,7 @@ AS_IF([test ! -d "${PIDPATH}" -o -h "${PIDPATH}" && test -d "/run" -a ! -h "/run
 
 AC_CHECK_PROGS([GETENT], [getent], [])
 AC_CHECK_PROGS([ID], [id], [])
+AC_CHECK_PROGS([LSGROUP], [lsgroup], [])
 
 PROBE_OS_USER="false"
 PROBE_OS_GROUP="false"
@@ -534,20 +535,24 @@ AS_IF([test x"${GETENT}" != x], [
 	],[
 	AS_IF([test x"${ID}" != x], [
 		PROBE_OS_USER="${ID} -u "
-		AS_IF([test -r "/etc/groups"], [
-			dnl FIXME: For command prefix usage like used in this
-			dnl  script, we might be better off defining a shell
-			dnl  function and referring to it. Grep would catch
-			dnl  also user names that have a group as secondary!
-			PROBE_OS_GROUP="cat /etc/groups | ${GREP} -w "
+		AS_IF([test x"${LSGROUP}" != x], [
+			PROBE_OS_GROUP="${LSGROUP} "
 			], [
-			dnl This shows groups of a USER with specified name!
-			dnl But to probe for cases where string names are
-			dnl same (e.g. "nut" or "ups") this might be good
-			dnl enough.
-			PROBE_OS_GROUP="${ID} -g "
+			AS_IF([test -r "/etc/groups"], [
+				dnl FIXME: For command prefix usage like used in this
+				dnl  script, we might be better off defining a shell
+				dnl  function and referring to it. Grep would catch
+				dnl  also user names that have a group as secondary!
+				PROBE_OS_GROUP="cat /etc/groups | ${GREP} -w "
+				], [
+				dnl This shows groups of a USER with specified name!
+				dnl But to probe for cases where string names are
+				dnl same (e.g. "nut" or "ups") this might be good
+				dnl enough.
+				PROBE_OS_GROUP="${ID} -g "
+				])
+			AC_MSG_WARN([Can not PROPERLY check existence of group accounts on this system, but can try best-effort])
 			])
-		AC_MSG_WARN([Can not PROPERLY check existence of group accounts on this system, but can try best-effort])
 		],[
 		AC_MSG_WARN([Can not check existence of user and group accounts on this system])
 		])

--- a/m4/nut_compiler_family.m4
+++ b/m4/nut_compiler_family.m4
@@ -111,6 +111,10 @@ dnl NOTE: this option should not be passed via the fourth argument of the macro,
 dnl or it ends up in the flags too, possibly during the "pop"; have to use the
 dnl GOOD_FLAG instead :\
     COMPILERFLAG="$1"
+dnl GCC will silently accept warning disable options, but then if there is an issue, warn about the option, i.e.
+dnl "cc1: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics"
+dnl Instead try the enable option to check if it is valid
+    COMPILERTESTFLAG="`echo $1 | sed 's/^-Wno-/-W/'`"
 
 dnl We also try to run an actual build since tools called from that might
 dnl complain if they are forwarded unknown flags accepted by the front-end.
@@ -119,14 +123,22 @@ dnl complain if they are forwarded unknown flags accepted by the front-end.
     AC_MSG_NOTICE([Starting check compile flag for '${COMPILERFLAG}'; now CFLAGS='${CFLAGS}' and CXXFLAGS='${CXXFLAGS}'])
 
     AC_LANG_PUSH([C])
+dnl Use a simple program with out previous #defines so we don't cause errors on clang with -Wreserved-identifier
+    m4_pushdef([AC_LANG_CONFTEST(C)], [m4_ifdef([AC_LANG_DEFINES_PROVIDED], [AC_LANG_DEFINES_PROVIDED])
+cat << _ACEOF > conftest.$ac_ext
+int main(void) { return 0; }
+_ACEOF
+    ])
     GOOD_FLAG=no
-    AX_CHECK_COMPILE_FLAG([${COMPILERFLAG}],
+    # CFLAGS="-Werror $CFLAGS"
+    AX_CHECK_COMPILE_FLAG([${COMPILERTESTFLAG}],
         [CFLAGS="-Werror $NUT_SAVED_CFLAGS ${COMPILERFLAG}"
          AC_MSG_CHECKING([whether the flag '${COMPILERFLAG}' is still supported in CC linker mode])
          AX_RUN_OR_LINK_IFELSE([AC_LANG_PROGRAM([],[])],
             [GOOD_FLAG=yes],[])
          AC_MSG_RESULT([${GOOD_FLAG}])
         ], [], [])
+    m4_popdef([AC_LANG_CONFTEST(C)])
     AC_LANG_POP([C])
     AS_IF([test x"${GOOD_FLAG}" = xyes],
         [CFLAGS="$NUT_SAVED_CFLAGS ${COMPILERFLAG}"],
@@ -135,14 +147,22 @@ dnl complain if they are forwarded unknown flags accepted by the front-end.
     AC_MSG_NOTICE([${GOOD_FLAG} for C '${COMPILERFLAG}'; now CFLAGS=${CFLAGS}])
 
     AC_LANG_PUSH([C++])
+dnl Use a simple program with out previous #defines so we don't cause errors on clang with -Wreserved-identifier
+    m4_pushdef([AC_LANG_CONFTEST(C++)], [m4_ifdef([AC_LANG_DEFINES_PROVIDED], [AC_LANG_DEFINES_PROVIDED])
+cat << _ACEOF > conftest.$ac_ext
+int main(void) { return 0; }
+_ACEOF
+    ])
     GOOD_FLAG=no
-    AX_CHECK_COMPILE_FLAG([${COMPILERFLAG}],
+    # CXXFLAGS="-Werror $CXXFLAGS"
+    AX_CHECK_COMPILE_FLAG([${COMPILERTESTFLAG}],
         [CXXFLAGS="-Werror $NUT_SAVED_CXXFLAGS ${COMPILERFLAG}"
          AC_MSG_CHECKING([whether the flag '${COMPILERFLAG}' is still supported in CXX linker mode])
          AX_RUN_OR_LINK_IFELSE([AC_LANG_PROGRAM([],[])],
             [GOOD_FLAG=yes],[])
          AC_MSG_RESULT([${GOOD_FLAG}])
         ], [], [])
+    m4_popdef([AC_LANG_CONFTEST(C++)])
     AC_LANG_POP([C++])
     AS_IF([test x"${GOOD_FLAG}" = xyes],
         [CXXFLAGS="$NUT_SAVED_CXXFLAGS ${COMPILERFLAG}"],

